### PR TITLE
Refactor oauth

### PR DESCRIPTION
**Pull request by qbalukom**

### CODE REVIEW

File: core-ui/src/components/Predefined/Create/OAuthClients/OAuthClients.create.js [hunk 1]
```diff
@@ -1,124 +1,177 @@
...
+      />
+      <TextArrayInput
+        advanced
+        propertyPath="$.spec.redirectUris"
+        title={t('oauth2-clients.labels.redirect-uris')}
+      />
+      <TextArrayInput
+        advanced
+        propertyPath="$.spec.postLogoutRedirectUris"
+        title={t('oauth2-clients.labels.post-logout-redirect-uris')}
```
Comment (Wawrzyn321) [inline core-ui/src/components/Predefined/Create/OAuthClients/OAuthClients.create.js:160]: Should we use here `type=url` or a `pattern` for input?

Failed to create the OAuth2 Client: OAuth2Client.hydra.ory.sh "aa" is invalid: 
* spec.postLogoutRedirectUris: Invalid value: "b": spec.postLogoutRedirectUris in body should match '\w+:/?/?[^\s]+', 
* spec.redirectUris: Invalid value: "a": spec.redirectUris in body should match '\w+:/?/?[^\s]+'].

**Review by Wawrzyn321 — CHANGES_REQUESTED** [review]
Just a validation comment.

**Review by Wawrzyn321 — DISMISSED** [review]

File: core/src/i18n/en.yaml [hunk 2]
```diff
@@ -1062,22 +1062,41 @@ node-details:
...
     code: Code
-    id-token: ID Token
+    id_token: ID Token
     token: Token
+  secret: OAuth2 Client Credentials
   title: OAuth2 Clients
+  tooltips:
+    audience: Audience is a whitelist defining the audiences this client is allowed to request tokens for.
+    client-name: Client name is the human-readable string name of the client to be presented to the end-user during authorization.
+    secret-name: Secret name points to the K8s secret that contains this client's ID and password.
```
Comment (grego952) [inline core/src/i18n/en.yaml:1099]:
```suggestion
    secret-name: Secret name points to the K8s Secret that contains this client's ID and password.
```

**Review by Wawrzyn321 — DISMISSED** [review]

File: core/src/i18n/en.yaml [hunk 3]
```diff
@@ -1062,22 +1062,41 @@ node-details:
...
     code: Code
-    id-token: ID Token
+    id_token: ID Token
     token: Token
+  secret: OAuth2 Client Credentials
   title: OAuth2 Clients
+  tooltips:
+    audience: Audience is a whitelist defining the audiences this client is allowed to request tokens for.
+    client-name: Client name is the human-readable string name of the client to be presented to the end-user during authorization.
+    secret-name: Secret name points to the K8s Secret that contains this client's ID and password.
```
Comment (klaudiagrz) [inline core/src/i18n/en.yaml:1099]:
```suggestion
    secret-name: Secret name points to the Kubernetes Secret that contains the client's ID and password.
```

**Review by klaudiagrz — APPROVED** [review]

**Review by Wawrzyn321 — APPROVED** [review]

**Pull request merged**